ewkcoffea/modules/corrections.py

```diff
@@ -140,69 +144,121 @@ def run3_muons_sf_Attach(muons,year,syst,id_method,iso_method):
+    # Finding the syst uncertaintities
+    sf_iso_flat_hi = sf_iso_flat_nom + sf_iso_flat_syst
+    sf_iso_flat_lo = sf_iso_flat_nom - sf_iso_flat_syst
+
+    # Getting the overall SF (ID * Iso)
+    sf_flat_nom = sf_id_flat_nom * sf_iso_flat_nom
+    sf_flat_hi = sf_id_flat_hi * sf_iso_flat_hi
+    sf_flat_lo = sf_id_flat_lo * sf_iso_flat_lo
+
+    # Unflatten to match the original structure
+    sf_nom = ak.unflatten(sf_flat_nom,ak.num(muons.pt))
+    sf_hi = ak.unflatten(sf_flat_hi,ak.num(muons.pt))
+    sf_lo = ak.unflatten(sf_flat_lo,ak.num(muons.pt))
+
+    muons['sf_nom_muon'] = sf_nom
+    muons['sf_hi_muon']  = sf_hi
+    muons['sf_lo_muon']  = sf_lo
+    muons['sf_nom_elec'] = ak.ones_like(sf_nom)
+    muons['sf_hi_elec']  = ak.ones_like(sf_nom)
+    muons['sf_lo_elec']  = ak.ones_like(sf_nom)
+
+
+def run3_electrons_sf_attach(electrons,year,wp):
 
     # Get the right sf json for the given campaign
     if year == "2022EE":
-        n_year = "2022Re-recoE+PromptFG"
-        fname = ewkcoffea_path("data/run3_lep_sf/electron_sf/electron.json")
+        n_year = "2022Re-recoE+PromptFG" # key for accessing the 2022EE SFs
+        fname = ewkcoffea_path("data/run3_lep_sf/electron_sf/2022EE_ele/electron.json")
     elif year == "2022":
-        #fname = ewkcoffea_path("data/run3_sf/muon_sf/ScaleFactors_Muon_Z_ID_ISO_2022_schemaV2.json")
-        raise Exception("Eras B,C,D, are not implemented yet!")
+        n_year = "2022Re-recoBCD" # key for accessing the 2022 SFs
+        fname = ewkcoffea_path("data/run3_lep_sf/electron_sf/2022_ele/electron.json")
     else:
         raise Exception("Trying to apply run3 SF where they shouldn't be!")
 
-    # Flatten the input (until correctionlib handles jagged data natively)
+    # Flatten the input
     eta_flat = ak.flatten(electrons.eta)
     pt_flat = ak.flatten(electrons.pt)
+    ceval = correctionlib.CorrectionSet.from_file(fname)
 
-    # For now, min pt sf is 10.0
-    pt_flat = ak.where(pt_flat<10.0,10.0,pt_flat)
+    # Create three pT regions (based on the json pT regions for the working points)
+    pt_flat_20 = ak.where(pt_flat >= 20.0,19.9,pt_flat)
+    pt_flat_2075 = ak.where(pt_flat < 20.0, 20.0, ak.where(pt_flat >= 75.0, 74.9, pt_flat))
+    pt_flat_75 = ak.where(pt_flat < 75.0, 75.0,pt_flat)
 
-    # Evaluate the SF
-    ceval = correctionlib.CorrectionSet.from_file(fname)
-    sf_flat = ceval["Electron-ID-SF"].evaluate(n_year,valtype,wp,eta_flat,pt_flat)
-    sf = ak.unflatten(sf_flat,ak.num(electrons.pt))
+    #Get the Reco SF for all three region lists
+    sf_flat_20 = ceval["Electron-ID-SF"].evaluate(n_year,"sf","RecoBelow20",eta_flat,pt_flat_20)
+    sf_flat_2075 = ceval["Electron-ID-SF"].evaluate(n_year,"sf","Reco20to75",eta_flat,pt_flat_2075)
+    sf_flat_75 = ceval["Electron-ID-SF"].evaluate(n_year,"sf","RecoAbove75",eta_flat,pt_flat_75)
 
-    electrons['ele_sf'] = sf
-    electrons['muon_sf'] = ak.ones_like(electrons.pt)
+    # Remove the unwanted Reco SF
```

Comment:
Just to save ourself from future confusion, can you be more detailed in this comment? E.g. something like "remove the SFs that go with electrons that were outside of the pt range applicable for these SFs" or something like that